cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java

```diff
@@ -600,4 +631,24 @@ protected int getOffset(GregorianCalendar calendar, Resolution resolution) {
         return offset;
     } // getOffset
 
+    /**
+     * Gets the mongo auth_source
+     * @return
+     */
+    public String getAuthSource() {
+        return mongoAuthSource;
+    } // getAuthSource
+
+    /**
+     * Set the mongo auth_source
+     * @param authSource DB used for mongo authentication
+     */
+    public void setAuthSource(String authSource) {
+        if (authSource != null) {
+            this.mongoAuthSource = authSource;
+        } else {
+            this.mongoAuthSource = "";
+        }
+    } // setAuthSource
+
```

Comment:
is this ever used?

Reply:
Again, if somebody or other piece of code (allowing other new sink in the future to use mongo as a DB with authentication),  and want to use the first constructor without authSource in parameter, it will have the opportunity to set the authSource after to build the backend object using the setAuthSource(...) operation when needed and in some place in the code. It is typical and best practices to do it in OOP (overload constructors and get/set operations), and give more possibilities.
Notice that in this case mongoAuthSource shoud not be annotated as "final" in the declaration.

Reply:
get/setAuthSource(...) could be used, if somebody build the object without authSource parameter and if after they want to initialize this parameter.
This is what I tested successfully and put in comment as documentation (see start() operation), and use of the new constructor.

BTW, What does it mean "NTC" and "LGTM"?

Reply:
NTC = Nothing to change
LGTM = Looks good to me

Reply:
More on "the NTC and LGTM protocol" review protocol here: https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/contribution_guidelines.md#pull-request-protocol